modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1978,13 +1980,109 @@ def sort_by(
+            columns = [columns]
+        # When we do a sort on the result of Series.value_counts, we don't rename the index until
+        # after everything is done, which causes an error when sorting the partitions, since the
+        # index and the column share the same name, when in actuality, the index's name should be
+        # None. This fixes the indexes name beforehand in that case, so that the sort works.
+
+        def sort_function(df):
+            index_renaming = None
+            if any(name in df.columns for name in df.index.names):
+                index_renaming = df.index.names
+                df.index = df.index.set_names([None] * len(df.index.names))
+            df = df.sort_values(by=columns, ascending=ascending, **kwargs)
+            if index_renaming is not None:
+                df.index = df.index.set_names(index_renaming)
+            return df
+
+        axis = Axis(axis)
+        if axis == Axis.ROW_WISE:
+            # If this df is empty, we don't want to try and shuffle or sort.
+            if len(self.axes[0]) == 0 or len(self.axes[1]) == 0:
+                return self.copy()
+            # If this df only has one row partition, we don't want to do a shuffle and sort - we can
+            # just do a full-axis sort.
+            if len(self._partitions) == 1:
+                return self.apply_full_axis(
+                    1,
+                    sort_function,
+                )
+            if self.dtypes[columns[0]] == object:
+                # This means we are not sorting numbers, so we need our quantiles to not try
+                # arithmetic on the values.
+                method = "inverted_cdf"
+            else:
+                method = "linear"
+            from .utils import build_sort_functions
+
+            shuffling_functions = build_sort_functions(
+                self,
+                columns[0],
+                method,
+                ascending if not isinstance(ascending, list) else ascending[0],
+                **kwargs,
+            )
+            major_col_partition_index = self.columns.get_loc(columns[0])
+            cols_seen = 0
+            index = -1
+            for i, length in enumerate(self._column_widths_cache):
+                cols_seen += length
+                if major_col_partition_index <= cols_seen:
+                    index = i
+                    break
+            new_partitions = self._partition_mgr_cls.shuffle_partitions(
+                self._partitions,
+                index,
+                shuffling_functions.sample_function,
+                shuffling_functions.pivot_function,
+                shuffling_functions.split_function,
+                sort_function,
+            )
+            new_axes = self.axes
+            if kwargs.get("ignore_index", False):
+                new_axes[axis.value] = RangeIndex(len(new_axes[axis.value]))
+            else:
+                new_axes[axis.value] = self._compute_axis_labels_and_lengths(
```

Comment:
we're modifying `self.axes` here, feels like a bug, maybe it should be a `new_axes = copy.copy(self.axes)` instead?

Reply:
The assignment here creates a copy (rather than a pointer), so when we modify new_axes, self.axes is unmodified! (Double checked using the code below):
```python
import modin.pandas as pd; import numpy as np
df = pd.DataFrame(np.random.randint(0, 100, (1000, 100)), columns=[f"col {i}" for i in range(100)])
mdf = df._query_compiler._modin_frame
print(mdf.axes)
a = mdf.axes
a[0] = 1
print(a)
print(mdf.axes)
```